gpflow/conditionals/util.py

```diff
@@ -54,6 +57,18 @@ def base_conditional(
     )  # [N]
     Kmn = tf.transpose(Kmn, perm)  # [..., M, N]
 
+    shape_constraints = [
+        (Kmn, [..., "M", "N"]),
```

Comment:
is this going to raise an error that contradicts the docstring, cos we're asserting on the altered shape?

Reply:
yes. and there's nothing we can do about it

Reply:
sure you can. you can add an explanation to the shape assertion explaining that the expected type isn't what the function expects

Reply:
I'd appreciate if you could suggest (in code) how you think that could look like!

Reply:
added